File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Gui/App/Models/Templates/TemplateCollaborationModel.ts
import TemplateCollaborationStatus = require('App/Models/Templates/TemplateCollaborationStatus');
import TemplateCollaborationRepresentationType = require('App/Models/Templates/TemplateCollaborationRepresentationType');
import EnumValue = require('App/Helpers/EnumValue');
import FolderPathValue = require('App/Helpers/FolderPathValue');
import TemplateCollaborationUserModel = require('App/Models/Templates/TemplateCollaborationUserModel');
import RoleModel = require('App/Models/Commons/RoleModel');

class TemplateCollaboration {
    UniqueId: string;
    Name: string;
    Status: TemplateCollaborationStatus;
    DocumentType: string;
    IdPriority: string;
    Object: string;
    Note: string;
    IsLocked: boolean;
    RegistrationUser: string;
    RegistrationDate: Date;
    JsonParameters: string;
    RepresentationType: TemplateCollaborationRepresentationType;
    TemplateCollaborationLevel: number;
    TemplateCollaborationPath: string;
    ParentInsertId: string;
    Roles: RoleModel[];
    TemplateCollaborationUsers: TemplateCollaborationUserModel[];

    public get StatusValue(): EnumValue {
        return new EnumValue(this.Status, TemplateCollaborationStatus);
    }

    public get RepresentationTypeValue(): EnumValue {
        return new EnumValue(this.RepresentationType, TemplateCollaborationRepresentationType);
    }

    public get TemplateCollaborationPathValue(): FolderPathValue {
        return new FolderPathValue(this.TemplateCollaborationPath);
    }
}

export = TemplateCollaboration;

File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Model/Entities/Templates/TemplateCollaborationModel.cs
using System;

namespace VecompSoftware.DocSuiteWeb.Model.Entities.Templates
{
    public class TemplateCollaborationModel
    {
        #region [ Constructor ]
        public TemplateCollaborationModel()
        {
        }
        #endregion

        #region [ Properties ]

        public Guid? UniqueId { get; set; }
        public string Name { get; set; }
        public string DocumentType { get; set; }
        public string IdPriority { get; set; }
        public TemplateCollaborationStatus Status { get; set; }
        public string Object { get; set; }
        public bool? IsLocked { get; set; }
        public string Note { get; set; }
        public bool WSDeletable { get; set; }
        public bool WSManageable { get; set; }
        public string RegistrationUser { get; set; }
        public DateTimeOffset? RegistrationDate { get; set; }
        public DateTimeOffset? LastChangedDate { get; set; }
        public string LastChangedUser { get; set; }

        #endregion
    }
}
